pkg/cmd/server/server.go

```diff
@@ -139,6 +153,7 @@ func (c *Config) Complete(ctx context.Context) (RunnableServer, error) {
 			return nil, fmt.Errorf("failed to create datastore: %w", err)
 		}
 	}
+	closers = append(closers, ds.Close)
```

Comment:
You'd actually want to append in the opposite order (putting ds.Close as the first element and then closers as the tail) to emulate the stack-like behavior of `defer`

Reply:
It's a good point, but I think it's not relevant because these `Close` methods are idempotent. I can still add it to simulate the defer unwinding.